spec/Section 2 -- Language.md

```diff
@@ -275,6 +277,90 @@ operations, each operation must be named. When submitting a Document with
 multiple operations to a GraphQL service, the name of the desired operation to
 be executed must also be provided.
 
+### Source schema document
+
+SourceSchemaDocument : TypeSystemDefinitionOrExtension+
+
+TypeSystemDefinitionOrExtension :
+
+- TypeSystemDefinition
+- TypeSystemExtension
+
+TypeSystemDefinition :
+
+- SchemaDefinition
+- TypeDefinition
+- DirectiveDefinition
+
+A {SourceSchemaDocument} describes the user types of a schema. A GraphQL
+implementation may use a {SourceSchemaDocument} as input to represent the schema
+and execute operations.
+
+For brevity, and to avoid conflicting with the implementation, all _built-in
+definitions_ must be omitted in a {SourceSchemaDocument}.
+
+**Default Root Operation Type Names**
+
+:: The _default root type name_ for each {`query`}, {`mutation`}, and
+{`subscription`} _root operation type_ are {"Query"}, {"Mutation"}, and
+{"Subscription"} respectively.
+
+The type system definition language can omit the schema definition when each
+_root operation type_ uses its respective _default root type name_ and no other
+type uses any _default root type name_.
+
+Likewise, when part of a {SourceSchemaDocument}, a schema definition should be
+omitted if each _root operation type_ uses its respective _default root type
+name_ and no other type uses any _default root type name_.
+
+This example describes a valid complete GraphQL schema, despite not explicitly
+including a {`schema`} definition. The {"Query"} type is presumed to be the
+{`query`} _root operation type_ of the schema.
+
+```graphql example
+type Query {
+  someField: String
+}
+```
+
+This example describes a valid GraphQL schema without a {`mutation`} _root
+operation type_, even though it contains a type named {"Mutation"}. The schema
+definition must be included, otherwise the {"Mutation"} type would be
+incorrectly presumed to be the {`mutation`} _root operation type_ of the schema.
+
+```graphql example
+schema {
+  query: Query
+}
+
+type Query {
+  latestVirus: Virus
+}
+
+type Virus {
+  name: String
+  mutations: [Mutation]
+}
+
+type Mutation {
+  name: String
+}
+```
+
+### Full schema document
+
+FullSchemaDocument : TypeSystemDefinition+
+
+A {FullSchemaDocument} describes all the types in the schema, including the
+built-in ones. Tools that are not implementations may use a {FullSchemaDocument}
+to validate an operation.
+
+All _built-in definitions_ must be included in a {SourceSchemaDocument}.
```

Comment:
FullSchemaDocument

Though I'm not sure this is true either. Any built-in definitions not included will be unavailable to tooling, and imply the schema **is not spec compliant** which is not the same as "must be included".

But for instance if there are no Float usages, does the full schema need to include the Float type?

Reply:
> FullSchemaDocument

Indeed, thanks!

> Though I'm not sure this is true either. Any built-in definitions not included will be unavailable to tooling, and imply the schema is not spec compliant which is not the same as "must be included".

Reason I went with this requirement is we're plagued with duplicate directive definitions. Latest example to date is https://github.com/apollographql/apollo-ios/issues/3287. Right now, we've been adding directive definitions in order to validate documents but this really fragile to cases where the directive definitions do not match like this one.

> if there are no Float usages, does the full schema need to include the Float type?

Good point. I guess it's not really required. The rule would then be "all used built-in definitions must be included", which is more flexible but also more prone to interpretation... But could be cool indeed.